### PR TITLE
[SDFAB-1140] Test DENY app filtering rule via QER gating in STC

### PR DESCRIPTION
**Pull request by daniele-moro**
Add step for app-filtering PFCP scenario, to test an app filtering rule with a closed gate in the QER.

Depends on:
- https://github.com/omec-project/pfcpsim/pull/57

### CODE REVIEW

File: .env.stable [hunk 1]
```diff
@@ -2,9 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-PFCP_AGENT_IMAGE=omecproject/upf-epc-pfcpiface:master-f3a2c33
+PFCP_AGENT_IMAGE=ghcr.io/omec-project/upf-epc/upf-epc-pfcpiface:pr-565
```
Comment (daniele-moro) [inline .env.stable:5]: Change when https://github.com/omec-project/upf/pull/565 is merged

File: .env.stable [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
 ONOS_IMAGE=opennetworking/sdfabric-onos:master-2022-03-05
-PFCPSIM_IMAGE=opennetworking/pfcpsim:1da9a55c
+PFCPSIM_IMAGE=dmoro92/pfcpsim:0.2.0-dev
```
Comment (daniele-moro) [inline .env.stable:7]: Change when https://github.com/omec-project/pfcpsim/pull/55 is merged

File: scenarios/pfcp-app-filtering.xml [hunk 3]
```diff
@@ -3,56 +3,98 @@
...
+                          exec="${DOCKER_COMPOSE_CMD} exec -T mock-smf pfcpctl session modify --count 5 --baseID ${#}0 --ue-pool 17.0.${#-1}.0/24 --gnb-addr 140.0.10${#-1}.1"/>
+                </sequential>
+            </group>
+            <step name="Check-Up4-Flows-App-Filtering-2" requires="Pfcp-Push-App-Filtering-2" delay="5"
+                  exec="onos-cli-grep ${OCI} up4:read-flows 'Apps=1, UL sess=${PFCP_UP4_FLOWS}, UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
+            <step name="Check-Flow-Rules-App-Filtering-2" requires="Pfcp-Push-App-Filtering-2" delay="5"
+                  exec="onos-check-flows ${OCI}"/>
+        </group>
+        <group name="Check-Traffic-App-Filtering-Negative-2" requires="Pfcp-Setup-App-Filtering-2" delay="5">
+            <parallel var="${ENODEB#}">
```
Comment (pierventre) [inline scenarios/pfcp-app-filtering.xml:80]: is it not dangerous doing these steps in parallel ?

Comment (daniele-moro) [reply scenarios/pfcp-app-filtering.xml:80]: Why do you think is dangerous? We send traffic and check if we didn't receive any packets.

Comment (pierventre) [reply scenarios/pfcp-app-filtering.xml:80]: I don't think you have any guarantee about the ordering - what about if you send first the traffic and then check the rcv ? You are not actually testing the behavior you expect - no ?

Comment (daniele-moro) [reply scenarios/pfcp-app-filtering.xml:80]: The steps sending traffic have a delay of 5 sec, this should save us from missing packets.

**Comment by daniele-moro** [comment]
retest this please

**Comment by daniele-moro** [comment]
retest this please

File: scenarios/pfcp-app-filtering.xml [hunk 4]
```diff
@@ -3,56 +3,104 @@
...
             </group>
-            <step name="Check-Up4-Flows-App-Filtering" requires="Pfcp-Push-App-Filtering" delay="5"
+            <step name="Check-Up4-Flows-App-Filtering-1" requires="Pfcp-Push-App-Filtering-1" delay="5"
                   exec="onos-cli-grep ${OCI} up4:read-flows 'Apps=1, UL sess=${PFCP_UP4_FLOWS}, UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
-            <step name="Check-Flow-Rules-App-Filtering" requires="Pfcp-Push-App-Filtering" delay="5"
+            <step name="Check-Flow-Rules-App-Filtering-1" requires="Pfcp-Push-App-Filtering-1" delay="5"
                   exec="onos-check-flows ${OCI}"/>
         </group>
-        <group name="Check-Traffic-App-Filtering-Negative" requires="Pfcp-Setup-App-Filtering" delay="5">
+        <group name="Check-Traffic-App-Filtering-Negative-1" requires="Pfcp-Setup-App-Filtering-1" delay="5">
```
Comment (pierventre) [inline scenarios/pfcp-app-filtering.xml:27]: I am confuse here - we are not setting any gating - why is it getting dropped ?

Comment (daniele-moro) [reply scenarios/pfcp-app-filtering.xml:28]: I hardcoded the port and added a comment explaining that we do a negative test for traffic not matching the application filter, that should be dropped.

**Review by pierventre — APPROVED** [review]

**Comment by daniele-moro** [comment]
retest this please

**Pull request merged**